solr/core/src/java/org/apache/solr/util/circuitbreaker/CPUCircuitBreaker.java

```diff
@@ -110,6 +113,27 @@ public double getCpuUsageThreshold() {
   }
 
   protected double calculateLiveCPUUsage() {
-    return operatingSystemMXBean.getSystemLoadAverage();
+    Metric metric = this.core
+        .getCoreContainer()
+        .getMetricManager()
+        .registry("solr.jvm")
+        .getMetrics()
+        .get("os.systemCpuLoad");
+
+    if (metric == null) {
+        return -1.0;
+    }
+
+    if (metric instanceof Gauge) {
+      @SuppressWarnings({"rawtypes"})
+          Gauge gauge = (Gauge) metric;
+      // unwrap if needed
+      if (gauge instanceof SolrMetricManager.GaugeWrapper) {
+        gauge = ((SolrMetricManager.GaugeWrapper) gauge).getGauge();
+      }
+      return ((Double) gauge.getValue()).doubleValue();
+    }
+
+    return -1.0;                // Unable to unpack metric
```

Comment:
Please log a warning here -- we should not silently project that the system is not under load

Reply:
There is a warning, see the documentation for details.

Reply:
I checked. The caller will look for <0 values and warn-log "Unable to get CPU usage". I don't fancy special return values to signal errors when there could be a dedicated Exception and catch logic though.